src/main/java/it/gov/pagopa/fdr/service/PaymentService.java

```diff
@@ -170,8 +170,14 @@ public GenericResponse addPaymentToExistingFlow(
     long numberOfAlreadyUsedIndexes =
         paymentRepository.countByFlowIdAndIndexes(publishingFlow.getId(), indexes);
     if (numberOfAlreadyUsedIndexes > 0) {
+      List<PaymentEntity> indexesAlreadyAdded =
+          paymentRepository.findByFlowIdAndIndexes(publishingFlow.getId(), indexes);
```

Comment:
Is the double query that differs only in count better? Probably yes

I am referring to these 2:
paymentRepository.countByFlowIdAndIndexes(publishingFlow.getId(), indexes);
paymentRepository.findByFlowIdAndIndexes(publishingFlow.getId(), indexes);

Reply:
Yes, the queries are the same. I prefer leave the `count(*)` because is more performant than the other and the conflicting zone is less probable to occur.